mods/features/preferredVideoQuality.js

```diff
@@ -0,0 +1,149 @@
+import { configRead, configChangeEmitter } from "../config.js";
+
+class PreferredQualityHandler {
+    constructor(videoID) {
+        this.videoID = videoID;
+        this.video = null;
+        this.attachVideoTimeout = null;
+        this.boundApplyQuality = this.applyQuality.bind(this);
+    }
+
+    init() {
+        this.attachVideo();
+
+        configChangeEmitter.addEventListener('configChange', (ev) => {
+            if (ev.detail && ev.detail.key === 'preferredVideoQuality') {
+                this.applyQuality();
+            }
+        });
+    }
+
+    attachVideo() {
+        clearTimeout(this.attachVideoTimeout);
+        this.attachVideoTimeout = null;
+
+        this.video = document.querySelector('video');
+        if (!this.video) {
+            this.attachVideoTimeout = setTimeout(() => this.attachVideo(), 100);
+            return;
+        }
+
+        this.video.addEventListener('loadedmetadata', this.boundApplyQuality);
+        this.video.addEventListener('playing', this.boundApplyQuality);
+
+        this.applyQuality();
+    }
+
+    destroy() {
+        if (this.attachVideoTimeout) clearTimeout(this.attachVideoTimeout);
+        if (this.video) {
+            this.video.removeEventListener('loadedmetadata', this.boundApplyQuality);
+            this.video.removeEventListener('playing', this.boundApplyQuality);
+        }
+    }
+
+    applyQuality() {
+        const quality = configRead('preferredVideoQuality');
+        console.log("[PreferredQuality] Applying quality:", quality);
+
+        if (!quality) return;
+
+        try {
+            const { player, app } = Object.values(window._yttv || {}).reduce((acc, obj) => {
+                if (obj && typeof obj.getAvailableQualityData === 'function') {
+                    acc.player = obj;
+                }
+                if (obj && obj.instance && typeof obj.instance.resolveCommand === 'function') {
+                    acc.app = obj.instance;
+                }
+                return acc;
+            }, { player: null, app: null });
```

Comment:
You can simply import `resolveCommand.js` for resolveCommand and use `document.querySelector('.html5-video-player')` as the player.